src/rust/logs-generator/src/main.rs

```diff
@@ -0,0 +1,436 @@
+        if bytes_per_s > 0 && current_bytes >= bytes_per_s * time_resolution {
+            continue;
+        }
+
+        // Stop generation if configured duration reached
+        if duration > no_duration && now - start > duration {
+            print(
+                verbose,
+                format!(
+                    "Logs generation finished after {} seconds",
+                    (now - start).as_secs()
+                ),
+            );
+            break;
+        }
+
+        // Stop generation if configured limit reached
+        if total_logs != 0 && count_logs >= total_logs {
+            break;
+        }
+
+        // Generate logs from patterns
+        for pattern in &patterns {
+            let log = build_log(&pattern, &wordlist, &mut counter);
+            let mut saved_logs: u64 = 0;
+
+            match &fd {
+                Some(x) => saved_logs = save_log(&x, &log),
+                None => {}
+            }
+
+            if saved_logs == 1 {
+                current_logs += 1;
+                current_bytes += log.len() as u64;
+                count_logs += 1;
+                count_bytes += log.len() as u64;
+            }
+
+            if total_logs != 0 && count_logs >= total_logs {
+                break;
+            }
+        }
+    }
+    print(verbose, format!("Sent {} logs in total", count_logs));
+    print(verbose, format!("Sent {} bytes in total", count_bytes));
+}
+
+fn save_log(mut fd: &File, log: &String) -> u64 {
+    let write = fd.write_all(log.as_bytes());
+    match write {
+        Ok(_result) => {
+            return 1;
+        }
+        Err(_err) => {
+            return 0;
+        }
+    }
+}
+
+fn read_wordlist(filename: &String) -> Vec<String> {
+    let mut vec = Vec::<String>::new();
+    let file = File::open(filename).unwrap();
+    for line in BufReader::new(file).lines() {
+        vec.push(line.unwrap());
+    }
+    return vec;
+}
+
+fn build_log(pattern: &String, wordlist: &Vec<String>, counter: &mut u64) -> String {
+    let mut rng = rand::thread_rng();
+    let slices = pattern.split_whitespace();
+    let mut log = "".to_owned();
+
+    for slice in slices {
+        if slice.starts_with("{") && slice.ends_with("}") {
+            // replace {w} with random word
+            if slice.contains("w") {
+                log += &get_random_word(wordlist);
+            }
+            // replace {d} with random digit
+            else if slice.contains("d") {
+                log += &rng.gen_range(0, 0xffffff).to_string();
+            }
+            // replace {c} with counter value
+            else if slice.contains("c") {
+                // counter
+                log += &counter.to_string();
+                *counter += 1;
+            }
+        } else {
+            log += slice;
+        }
+        log += " ";
+    }
+    log += "\n";
+    return log;
+}
+
+fn get_random_word(wordlist: &Vec<String>) -> &String {
+    let mut rng = rand::thread_rng();
```

Comment:
Not really so important but I guess this can somehow be initialized just once? Are there static variables in rust?

Reply:
```
error[E0133]: use of mutable static is unsafe and requires unsafe function or block
```